src/Atc.Azure.Messaging/EventHub/EventHubCredentialsPublisherFactory.cs

```diff
@@ -0,0 +1,36 @@
+using Atc.Azure.Options.Authorization;
+using Atc.Azure.Options.Environment;
+using Atc.Azure.Options.EventHub;
+using Atc.Azure.Options.Providers;
+using Azure.Identity;
+using Azure.Messaging.EventHubs.Producer;
+
+namespace Atc.Azure.Messaging.EventHub;
+
+#pragma warning disable CA2000 // Dispose objects before losing scope
```

Comment:
I think this should be restored at the end of the file? Or use the SuppressMessage attribute instead?

Reply:
I prefer to use the `SuppressMessage` on the method level

Reply:
fixed